samples: nrf5340: extxip_smp_svr: Convert to sysbuild (#16513)
nrfconnect/sdk-nrf: doc/nrf/app_dev/device_guides/nrf53/qspi_xip_guide_nrf5340.rst

   zephyr_code_relocate(LIBRARY subsys__mgmt__mcumgr__mgmt LOCATION EXTFLASH_TEXT NOCOPY)
   zephyr_code_relocate(LIBRARY subsys__mgmt__mcumgr__mgmt LOCATION RAM_DATA)

Building the project
********************

You must use child image when building your project, as XIP QSPI does not currently support sysbuild.

> 

Author: Fixed

> ********************

Author: Fixed

> For the nRF5340 DK and other boards equipped with flash working in the QSPI mode, use the ``west flash`` command.

Author: Fixed

> For other cases, a nrfjprog configuration file needs setting up.

Author: Fixed

Flashing to external flash in SPI/DSPI mode

> ===========================================

Author: Fixed

> Flashing an application with ``west`` triggers the ``nrfjprog`` runner.

Author: Fixed

> The runner uses the default system settings that configure the application in the QSPI mode when flashing the external flash.

Reviewer: ```suggestion
The runner uses the default system settings that configure the application in the QSPI mode when programming the external flash.
```

Author: Fixed

You can change this behavior by using a custom :file:`Qspi.ini` configuration file.

.. note::
    The :file:`Qspi.ini` file is required to work on the Nordic Thingy:53.

> This file can be created which specifies the mode to use when programming the QSPI flash, the following is for the Thingy:53 and uses ``PP`` for programming and ``READ2IO`` for reading:

Reviewer: ```suggestion
This file can specify the mode to use when programming the QSPI flash.
For example, the following code is from the file for the Thingy:53 and uses ``PP`` for programming and ``READ2IO`` for reading:
```

Author: Fixed

.. literalinclude:: ../../../../../samples/nrf5340/extxip_smp_svr/Qspi_thingy53.ini
    :language: yaml

> To use this file automatically in a project, the ``CMakeLists.txt`` file needs updating, this can be applied globally for every board built with a project or applied to specific boards if a project supports multiple board targets, each with different configurations.

Reviewer: ```suggestion
To use this file automatically in a project, update the ``CMakeLists.txt`` file by :ref:`adding the mention of the new file <modifying_files_compiler>`.
This way, the file can be applied globally for every board built with a project or applied to specific boards if a project supports multiple board targets, each with different configurations.
```

Author: Fixed

> The following shows how to set the configuration file used when flashing the Thingy:53 only, note that this code **must** be placed before the ``find_package`` line:

Reviewer: ```suggestion
The following code shows how to set the configuration file used when flashing the Thingy:53 only.

.. note::
    This code must be placed before the ``find_package`` line.
```

Author: Fixed

+-----------------+-----------------+--------+--------------+--------+-----------+--------------------+--------------------+--------------------------+--------------------------+
| 128 MHz         | External flash  | Yes    | 48 MHz       | Quad   | 36.4      | 8.85               | 13.9               | 966                      | 911                      |
+-----------------+-----------------+--------+--------------+--------+-----------+--------------------+--------------------+--------------------------+--------------------------+

Additional information
**********************

> Additional information regarding the QSPI XIP mode of |NCS| and how to utilize it can be found on :ref:`qspi_xip_split_image`.

Reviewer: ```suggestion
For additional information regarding the QSPI XIP mode of |NCS| and how to use it, see :ref:`qspi_xip_split_image`.
```

Author: Fixed
